corehq/apps/app_manager/dbaccessors.py

```diff
@@ -178,16 +178,20 @@ def get_app_cached(domain, app_id):
     which are read-only.
 
     This only caches app builds."""
-    key = 'app_build_cache_{}_{}'.format(domain, app_id)
-    app = cache.get(key)
-    if not app:
-        app = get_app(domain, app_id)
-        if app.copy_of:
-            cache.set(key, app, 24 * 3600)
-
+    app = get_app_cached()
+    if not app.copy_of:
+        _get_app_cached.clear(app_id)
     return app
 
 
+@quickcache(['domain', 'app_id'], timeout=24 * 3600,
+    memoize_timeout=24 * 3600, session_function=None)
```

Comment:
Should we do this only for ICDS since on other envs this might drain RAM of the app workers given that this will be called for a lot of apps whereas an ICDS only few apps?